rollout/controller.go

```diff
@@ -228,7 +228,18 @@ func NewController(cfg ControllerConfig) *Controller {
 	log.Info("Setting up event handlers")
 	// Set up an event handler for when rollout resources change
 	cfg.RolloutsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: controller.enqueueRollout,
+		AddFunc: func(obj interface{}) {
+			controller.enqueueRollout(obj)
+			ro := unstructuredutil.ObjectToRollout(obj)
+			if ro != nil && cfg.Recorder != nil {
+				cfg.Recorder.Eventf(ro, record.EventOptions{
+					EventType:   corev1.EventTypeNormal,
+					EventReason: conditions.RolloutAddedToInformerReason,
+				}, "Rollout resource added to informer: %s/%s", ro.Namespace, ro.Name)
+			} else {
+				log.Warnf("Malformed rollout resource added to informer")
```

Comment:
I am afraid that this message can be misleading if the `cfg.Recorder` is nil. Also, it seems that `unstructuredutil.ObjectToRollout` is already log warning if it fails to convert.

Reply:
Changed it up a bit to depend on the log from ObjectToRollout for ro=nil and handled the Recorder being nil log